src/IPAddr.cc

```diff
@@ -24,46 +24,115 @@ namespace detail
 
 ConnKey::ConnKey(const IPAddr& src, const IPAddr& dst, uint16_t src_port, uint16_t dst_port,
```

Comment:
The `ConnKey` changes come a bit surprising in this commit. What's the gist of the change there? (Hard to see from the diff without digging quite a bit).

Reply:
The Teredo analyzer's `remove_teredo_analyzer` method needed a way to generate a `ConnKey` from a `Val*`, since that's what the `conn_id` type in script-land is. We already had code to do that, but it was part of `session::Manager::FindConnection()`, not part of `ConnKey` itself. The big changes here move that code out into a new constructor that takes a Val*, refactor some of the common initialization code used in all of the constructors into a single method, and then call that new constructor in the places where it's needed.